rust/src/storage/file.rs

```diff
@@ -59,4 +60,23 @@ impl StorageBackend for FileStorageBackend {
             })
         })))
     }
+
+    async fn put_obj(&self, path: &str, obj_bytes: &[u8]) -> Result<(), StorageError> {
+        let dir = std::path::Path::new(path);
+        let dir = dir.parent().unwrap();
```

Comment:
we should propagate the error instead of doing unwrap here. basically avoid unwrap at all costs :)

Reply:
sort of fixed - used an `if let ...` guard to avoid the unwrap